### PR TITLE
TASK: Update to vimeo/psalm 4.1

### PR DESCRIPTION
**Pull request by kdambekalns**
See #2243

### CODE REVIEW

File: psalm.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0"?>
 <psalm
-    totallyTyped="false"
```
Comment (kdambekalns) [inline psalm.xml:3]: This is the default, thus not needed.

File: psalm.xml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xmlns="https://getpsalm.org/schema/config"
-    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    xsi:schemaLocation="https://getpsalm.org/schema/config ../Libraries/vimeo/psalm/config.xsd"
```
Comment (kdambekalns) [inline psalm.xml:5]: Makes it work in PhpStorm - and fits our file structure.

File: psalm.xml [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
     errorBaseline="Packages/Framework/psalm-baseline.xml"
+    resolveFromConfigFile="false"
```
Comment (kdambekalns) [inline psalm.xml:7]: Without this, the paths were not resolved from "here", even though it's supposed to be the default. 🤷‍♂️

File: psalm.xml [hunk 4]
```diff
@@ -38,7 +42,6 @@
         <MissingReturnType errorLevel="info" />
         <MissingPropertyType errorLevel="info" />
         <InvalidDocblock errorLevel="info" />
-        <MisplacedRequiredParam errorLevel="info" />
```
Comment (kdambekalns) [inline psalm.xml:41]: This no longer exists in Psalm 4.1

**Review by albe — APPROVED** [review]

**Pull request merged**